rct_optimizations/test/include/rct_optimizations_tests/pose_generator.h

```diff
@@ -0,0 +1,43 @@
+#ifndef POSE_GEN_H
+#define POSE_GEN_H
+
+#include "rct_optimizations_tests/observation_creator.h"
+#include <rct_optimizations_tests/utilities.h>
+#include <Eigen/Geometry>
+#include <vector>
+
+namespace rct_optimizations
+{
+namespace test
+{
+/**
+ * @brief genHemispherePose Generates camera poses in a hemisphere pattern
+ * @param target_pose The position/orientation of the target
+ * @param r Radius of the hemisphere
+ * @param theta_cnt The number of points in the theta-wise direction
+ * @param phi_cnt The number of points in the phi-wise direction
+ * @return A vector of camera positions & orientations
+ */
+std::vector<Eigen::Isometry3d> genHemispherePose(Eigen::Isometry3d target_pose, double r, int theta_cnt, int phi_cnt);
```

Comment:
`Eigen::Isometry3d` -> `const Eigen::Isometry3d&` and make your inputs all `const`